src/bootstrap_handler.rs

```diff
@@ -102,8 +92,10 @@ impl BootstrapHandler {
     }
 
     pub fn read_bootstrap_file(&mut self) -> io::Result<(Contacts)> {
-        self.file_path = BootstrapHandler::get_file_path();
-        let mut file = try!(File::open(&self.file_path));
+        let mut file = match File::open(&self.file_path) {
```

Comment:
Not caused by this PR, but this function doesn't `self` to be `mut`.  The same is true for `oldest_contacts` and `get_serialised_contacts`.